File: messageGo/grpc/baseData/team.go
package base_data

type TeamMember struct {
	UserId int64  `json:"userId"`
	Name   string `json:"name"`
	Icon   string `json:"icon"`
}

func (p *TeamMember) Clear() {
	p.UserId = 0
	p.Name = ""
	p.Icon = ""
}

type Team struct {
	Id      int64        `json:"id"`
	Name    string       `json:"name"`
	Leader  int64        `json:"leader"`
	Members []TeamMember `json:"members"`
}

func (p *Team) Clear() {
	p.Id = 0
	p.Name = ""
	p.Leader = 0
	for idx, _ := range p.Members {
		p.Members[idx].Clear()
	}
	p.Members = p.Members[:0]
}

File: messageDotNet/grpc/baseData/Team.cs
using System;
using UnityEngine;


[Serializable]
public class TeamMember
{
    public long UserId;
    public string Name;
    public string Icon;
}

[Serializable]
public class Team
{
    public long Id;
    public string Name;
    public long Leader;
    public TeamMember[] Members;
}
